File: CS50X_TS/PSET6/sentimental-cash/cash.py
from cs50 import get_float

# Prompt user for valid input
while True:
    a = get_float("Change: ")
    if a > 0:
        break

# Convert dollars to cents and round
cents = round(a * 100)

# Coin counter
coin = 0

# Coin values in cents
coin_values = [25, 10, 5, 1]

# Calculate the number of coins
for value in coin_values:
    coin += cents // value  # Add number of coins for current denomination
    cents %= value  # Update remaining cents

# Print total number of coins
print(coin)

File: CS50X_TS/PSET1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt the user for the amount of change owed in cents
    int cents;
    do
    {
        // Prompt the user for cents (not dollars)
        cents = get_int("Change owed: ");
    }
    while (cents < 0); // Repeat until a non-negative amount is entered

    // Initialize the coin count
    int coins = 0;

    // Calculate the number of 25-cent coins (quarters)
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }

    // Calculate the number of 10-cent coins (dimes)
    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }

    // Calculate the number of 5-cent coins (nickels)
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }

    // Calculate the number of 1-cent coins (pennies)
    while (cents >= 1)
    {
        cents -= 1;
        coins++;
    }

    // Print the total number of coins needed
    printf("%d\n", coins);
}
